core/celery/tasks/generate_inventory_and_hold_reports.py

```diff
@@ -0,0 +1,292 @@
+from __future__ import annotations
+
+import csv
+import os
+import tempfile
+from datetime import datetime
+from pathlib import Path
+from typing import Any
+
+from celery import shared_task
+from sqlalchemy import not_, select, text
+from sqlalchemy.orm import Session, sessionmaker
+
+from api.integration.registry.license_providers import LicenseProvidersRegistry
+from core.celery.job import Job
+from core.celery.task import Task
+from core.integration.goals import Goals
+from core.model import (
+    IntegrationConfiguration,
+    IntegrationLibraryConfiguration,
+    Library,
+    get_one,
+)
+from core.opds_import import OPDSImporterSettings
+from core.service.celery.celery import QueueNames
+from core.service.container import Services, container_instance
+
+
+class GenerateInventoryAndHoldsReportsJob(Job):
+    _delete_attachments = True
+
+    def __init__(
+        self, session_maker: sessionmaker[Session], library_id: int, email_address: str
+    ):
+        super().__init__(session_maker)
+        self.library_id = library_id
+        self.email_address = email_address
+
+    @property
+    def services(self) -> Services:
+        return container_instance()
```

Comment:
Instead of loading the container instance here in the job, can you pass the `send_email` function into `GenerateInventoryAndHoldsReportsJob` constructor from the task? Loading the container this way works... but I'd like to try not to have the Jobs depend directly on the container if we can help it.